### ISSUE TITLE
return beta-binomial dispersion parameters from DMLfit.multiFactor

### ISSUE THREAD

**Issue by dariomel** [haowulab/DSS#12]
currently, the model fitting results returned by `DMLfit.multiFactor` include the regression coefficients and their covariance matrices. It would be very useful if this function returned also the beta-binomial dispersion estimated at each DML.

**Comment by haowulab** [comment]
Good idea. I just modified the code to output the estimated dispersions. The changes will be reflects on bioc after a little while, but the new package source is attached here. 

[DSS_2.37.0.tar.gz](https://github.com/haowulab/DSS/files/4899272/DSS_2.37.0.tar.gz)

**Comment by dariomel** [comment]
thank you very much.
I just started using this new feature to run some simulations and it works well.

When I call `DMLfit.multiFactor` I get the following:
```R
Warning message:
In if ((!is.matrix(Y0) | !is.matrix(N0)) & (class(Y0) != "DelayedMatrix" |  :
  the condition has length > 1 and only the first element will be used
```
which I believe comes from line 49 in `DML.multiFactor.R`,
```R
    if( (!is.matrix(Y0) | !is.matrix(N0)) & (class(Y0)!="DelayedMatrix" | class(N0)!="DelayedMatrix") )
```

You might want to fix it by using `all(...)` around each comparison involving `class(...)`.

**Comment by haowulab** [comment]
This is strange. I never encountered this. The expression inside if should return a single boolean. Can you try the example for DMLfit.multiFactor to see what you get?

**Comment by ChristophH** [comment]
`class(x)` may return a vector if x inherits from multiple classes, you might want to use `inherits(x, what)` instead. e.g. `inherits(x = Y0, what = "DelayedMatrix")` instead of `class(Y0)!="DelayedMatrix"`

**Comment by dariomel** [comment]
The warning message in question shows up in Step 2 of [The DSS User's Guide, Section 3.4.2](http://bioconductor.org/packages/release/bioc/vignettes/DSS/inst/doc/DSS.html#342_example_analysis_for_data_from_general_experimental_design):

> Stepp 2. Fit a linear model using DMLfit.multiFactor function, include case, cell, and case by cell interaction. Similar to in a multiple regression, the model only needs to be fit once, and then the parameters can be tested based on the model fitting results.
> 
> ```
> DMLfit = DMLfit.multiFactor(RRBS, design=design, formula=~case+cell+case:cell)
> 
> ## Warning in if ((!is.matrix(Y0) | !is.matrix(N0)) & (class(Y0) != "DelayedMatrix"
> ## | : the condition has length > 1 and only the first element will be used
> ```

You may want to fix the spelling error too.

**Comment by haowulab** [comment]
I just removed the parts of checking whether they are DelayedArray. That should fix the problem.

I used to use DelayedArray in DSS so that it can handle large dataset better. But I found that makes the computation significantly slower, so I changed back. Now there's no DelayedArray so I don't need to check it.

**Comment by haowulab** [comment]
Moreover, I didn't realize it appears in the manual. I don't have that problem on my computer, running R 3.6.
